File: android/src/main/kotlin/io/bucketeer/sdk/flutter/CallMethods.kt
package io.bucketeer.sdk.flutter

import androidx.annotation.VisibleForTesting

internal enum class CallMethods {
  Initialize,
  StringVariation,
  StringVariationDetails,
  IntVariation,
  IntVariationDetails,
  DoubleVariation,
  DoubleVariationDetails,
  BoolVariation,
  BoolVariationDetails,
  JsonVariation,
  ObjectVariation,
  ObjectVariationDetails,
  Track,
  CurrentUser,
  UpdateUserAttributes,
  FetchEvaluations,
  Flush,
  EvaluationDetails,
  AddProxyEvaluationUpdateListener,
  Destroy,
  Unknown;

  companion object {
    private val avoidInitializationAssertions = listOf(
      Initialize,
      Destroy
    )

    fun shouldAssertInitialize(methods: CallMethods): Boolean {
      return !avoidInitializationAssertions.contains(methods);
    }
  }
}

File: lib/src/call_methods.dart
enum CallMethods {
  initialize,
  stringVariation,
  intVariation,
  doubleVariation,
  boolVariation,
  jsonVariation,
  track,
  currentUser,
  updateUserAttributes,
  fetchEvaluations,
  flush,
  evaluationDetails,
  addProxyEvaluationUpdateListener,
  destroy,
  unknown,

  objectVariation,
  objectVariationDetails,
  intVariationDetails,
  boolVariationDetails,
  doubleVariationDetails,
  stringVariationDetails,
}

abstract class CallMethodParams {
  static const featureId = 'featureId';
  static const defaultValue = 'defaultValue';
}
